src/components/tooltip/tooltip.pw.tsx

```diff
@@ -0,0 +1,375 @@
+      mount,
+      page,
+    }) => {
+      await mount(
+        <TooltipComponent isPartOfInput inputSize={inputSize} position="top" />
+      );
+
+      const element = getDataElementByValue(page, "tooltip");
+      await assertCssValueIsApproximately(element, "top", offset);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 5],
+    [SIZE.MEDIUM, 6],
+    [SIZE.LARGE, 10],
+  ] as [TooltipProps["inputSize"], number][]).forEach(([inputSize, offset]) => {
+    test(`should have correct styles applied when inputSize is ${inputSize} and tooltip position is set as bottom`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <TooltipComponent
+          isPartOfInput
+          inputSize={inputSize}
+          position="bottom"
+        />
+      );
+
+      const buttonRect = await page
+        .getByRole("button")
+        .evaluate((element) => element.getBoundingClientRect());
+
+      const element = getDataElementByValue(page, "tooltip");
+      await assertCssValueIsApproximately(
+        element,
+        "top",
+        buttonRect.top + buttonRect.height + offset
+      );
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 47],
+    [SIZE.MEDIUM, 44],
+    [SIZE.LARGE, 40],
+  ] as [TooltipProps["inputSize"], number][]).forEach(([inputSize, offset]) => {
+    test(`should have correct styles applied when inputSize is ${inputSize} and tooltip position is set as left`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <TooltipComponent isPartOfInput inputSize={inputSize} position="left" />
+      );
+
+      const element = getDataElementByValue(page, "tooltip");
+      await assertCssValueIsApproximately(element, "left", offset);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 5],
+    [SIZE.MEDIUM, 8],
+    [SIZE.LARGE, 12],
+  ] as [TooltipProps["inputSize"], number][]).forEach(([inputSize, offset]) => {
+    test(`should have correct styles applied when inputSize is ${inputSize} and tooltip position is set as right`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <TooltipComponent
+          isPartOfInput
+          inputSize={inputSize}
+          position="right"
+        />
+      );
+
+      const buttonRect = await page
+        .getByRole("button")
+        .evaluate((element) => element.getBoundingClientRect());
+
+      const element = getDataElementByValue(page, "tooltip");
+      await assertCssValueIsApproximately(
+        element,
+        "left",
+        buttonRect.left + buttonRect.width + offset
+      );
+    });
+  });
+
+  test(`should show tooltip when target is hovered`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<UncontrolledTooltipComponent />);
+
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+    const tooltipElement = page.getByRole("button");
+    await tooltipElement.hover();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+  });
+
```

Comment:
I think there are couple of tests are missing comparing with original cypress tests:` "should hide tooltip when mouse leaves target", "should show tooltip when target is focused", "should hide tooltip when target is blurred", "new tooltip target should still trigger tooltip visibility"`, is there any reason to remove them? If there's or anything I missed just ignore me